portal/views/portal.py

```diff
@@ -442,15 +439,28 @@ def challenge_identity(
         authenicated WRITE_ONLY invite account
     :param access_on_verify: boolean value, set true IFF on success, the
         user should be logged in once validated, i.e. w/o a password
+    :param request_path: the requested url prior to redirection to here
+        necessary in no cookie situations, to redirect user back
 
     """
+    # At this point, we can expect a session, or the user likely
+    # doesn't have cookies enabled.  (ignore misleading `_fresh`
+    # and `_permanent` keys)
+    session_keys = [k for k in session if k not in ('_fresh', '_permanent')]
+    if not session_keys:
+        request_path = request.args.get('request_path', request_path)
+        return redirect(url_for(
```

Comment:
@pbugni Should we have some kind of logging when this happens (info/debug)?

Reply:
sure.  good idea